### PR TITLE
On-chain TX source tracking

### PR DESCRIPTION
**Pull request by nick**

### CODE REVIEW

File: libs/shared/providers/src/txTracker/utils.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import { prepareWriteContract as prepareWriteContractOrig } from '@wagmi/core';
+import { keccak256, toBytes } from 'viem';
+
+const defaultSuffix = keccak256(toBytes('oeth.com')).slice(2, 10); // 9fed593b
```
Comment (toniocodo) [inline libs/shared/providers/src/txTracker/utils.ts:5]: Maybe pass the default suffix as a param to the hook so this can be reused on other apps

**Review by toniocodo — COMMENTED** [review]
It looks good to me, only have 2 suggestions:
- add a wrapper for the prepare hook `usePrepareContractWrite` (used [here](https://github.com/OriginProtocol/origin-defi/blob/cd1ea70ffbd87666ecc4da5acd64fabb262bf692/libs/shared/providers/src/rebaseBanner/components/RebaseBanner.tsx#L22))
- add a return value tx-trackId to `useTrackerTx` so that it can be manually pass

File: libs/shared/providers/src/txTracker/utils.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```
Comment (toniocodo) [inline libs/shared/providers/src/txTracker/utils.ts:1]: not sure it's necessary, which rule is it complaining about?

File: libs/shared/providers/src/txTracker/utils.ts (hunk 1 above)
Comment (nick) [reply libs/shared/providers/src/txTracker/utils.ts]: 👍

**Comment by nick** [comment]
Sounds good, I pushed some changes to address those comments

**Review by toniocodo — APPROVED** [review]
LGTM!

**Pull request merged**